python/ray/tests/test_multiprocessing.py

```diff
@@ -508,6 +508,31 @@ def f(args):
         result_iter.next()
 
 
+def test_fail_on_non_iterable(pool):
+    def fn(*_):
+        pass
+
+    non_iterable = 3
+
+    with pytest.raises(TypeError, match="object is not iterable"):
+        pool.imap(fn, non_iterable)
+
+    with pytest.raises(TypeError, match="object is not iterable"):
+        pool.imap_unordered(fn, non_iterable)
+
+    with pytest.raises(TypeError, match="object is not iterable"):
+        pool.map(fn, non_iterable)
+
+    with pytest.raises(TypeError, match="object is not iterable"):
+        pool.map_async(fn, non_iterable)
+
+    with pytest.raises(TypeError, match="must be an iterable, not"):
+        pool.starmap(fn, [non_iterable])
+
+    with pytest.raises(TypeError, match="must be an iterable, not"):
```

Comment:
where are these error messages coming from, the `iter` call? 

Not sure what it looks like, but should we instead wrap it and write a more user-friendly error message?